lib/ascii_binder/engine.rb

```diff
@@ -0,0 +1,872 @@
+        # switch to the old one and warn the user.
+        use_file = BUILD_FILENAME
+        warning "'#{BUILD_FILENAME}' is a deprecated filename. Rename this to '#{TOPIC_MAP_FILENAME}'."
+      end
+      use_file
+    end
+
+    def dir_empty?(dir)
+      Dir.entries(dir).select{ |f| not f.start_with?('.') }.empty?
+    end
+
+    # Protip: Don't cache this! It needs to be reread every time we change branches.
+    def build_config
+      validate_config(YAML.load_stream(open(File.join(source_dir,build_config_file))))
+    end
+
+    def create_new_repo
+      gem_template_dir = File.join(gem_root_dir,"templates")
+
+      # Create the new repo dir
+      FileUtils.mkdir_p(source_dir)
+
+      # Copy the basic repo content into the new repo dir
+      Find.find(gem_template_dir).each do |path|
+        next if path == gem_template_dir
+        src_path = Pathname.new(path)
+        tgt_path = src_path.sub(gem_template_dir,source_dir)
+        if src_path.directory?
+          FileUtils.mkdir_p(tgt_path.to_s)
+        else
+          FileUtils.cp src_path.to_s, tgt_path.to_s
+        end
+      end
+
+      # Initialize the git repo
+      Git.init(source_dir)
+    end
+
+    def find_topic_files
+      file_list = []
+      Find.find(source_dir).each do |path|
+        # Only consider .adoc files and ignore README, and anything in
+        # directories whose names begin with 'old' or '_' (underscore)
+        next if path.nil? or not path =~ /.*\.adoc/ or path =~ /README/ or path =~ /\/old\// or path =~ /\/_/
+        src_path = Pathname.new(path).sub(source_dir,'').to_s
+        next if src_path.split('/').length < 3
+        file_list << src_path
+      end
+      file_list.map{ |path|
+        parts = path.split('/').slice(1..-1);
+        parts.slice(0..-2).join('/') + '/' + parts[-1].split('.')[0]
+      }
+    end
+
+    def remove_found_config_files(branch,branch_build_config,branch_topic_files)
+      nonexistent_topics = []
+      branch_build_config.each do |topic_group|
+        tg_dir = topic_group['Dir']
+        topic_group['Topics'].each do |topic|
+          if topic.has_key?('File')
+            topic_path = tg_dir + '/' + topic['File']
+            result     = branch_topic_files.delete(topic_path)
+            if result.nil?
+              nonexistent_topics << topic_path
+            end
+          elsif topic.has_key?('Dir')
+            topic_path = tg_dir + '/' + topic['Dir'] + '/'
+            topic['Topics'].each do |subtopic|
+              result = branch_topic_files.delete(topic_path + subtopic['File'])
+              if result.nil?
+                nonexistent_topics << topic_path + subtopic['File']
+              end
+            end
+          end
+        end
+      end
+      if nonexistent_topics.length > 0
+        nl_warning "The #{build_config_file} file on branch '#{branch}' references nonexistant topics:\n" + nonexistent_topics.map{ |topic| "- #{topic}" }.join("\n")
```

Comment:
nonexistant is not a word. You used it correctly directly before and directly after.